sdv/metadata/single_table.py

```diff
@@ -645,8 +647,20 @@ def _validate_column_relationship(self, relationship_type, column_names):
                 errors.append(
                     f"Cannot use primary key '{column}' in column relationship."
                 )
+
+        columns_to_sdtypes = {
+            column: self.columns.get(column, {}).get('sdtype') for column in column_names
+        }
         try:
-            self._COLUMN_RELATIONSHIP_TYPES[relationship_type](self.columns, column_names)
+            self._COLUMN_RELATIONSHIP_TYPES[relationship_type](columns_to_sdtypes)
+        except ImportError:
+            warnings.warn(
+                f"The metadata contains a column relationship of type '{relationship_type}'. "
+                f'which requires the {relationship_type} add-on.'
+                'This relationship will be ignored. For higher quality data in this'
+                ' relationship, please inquire about the SDV Enterprise tier.')
+            del self.column_relationships[index]
```

Comment:
Not sure if we should actually delete it or just ignore it and leave it. @npatki What do you think? If a user provides a relationship type that they do not have access too (eg. because they don't have enterprise), should we delete it from their metadata or just ignore?

Reply:
@amontanez24 we should not delete it.

The metadata is considered a factual, ground source-of-truth. It should be correct regardless of which synthesizers/RDT features you have access to. Or another way to put it: When you get access to SDV Enterprise, you shouldn't have to go back and add to your metadata.

I understand that this is happening in-memory, but it is strange and unexpected behavior for us to be modifying the in-memory metadata object without the user's knowledge. This can lead to some nasty bugs down the line.